### ISSUE TITLE
Incorrect error location in .bess files

### ISSUE THREAD

**Issue by nemethf** [NetSys/bess#408]
Consider the following .bess file:
```python
Source() -> Bypass() -> Sink()

# arbitrary large number of correct python commands
a = 1
b = 2

print(a, b))  # Extra closing parenthesis
```
It has a syntax error, but bessclt points to an incorrect location, which makes hard to pinpoint the cause of the error.
```
$ ./bessctl daemon start -- run my/bug
Traceback (most recent call last):
  File "./bessctl", line 164, in <module>
    main()
  File "./bessctl", line 161, in main
    run_cli(cStringIO.StringIO('\n'.join(cmds)))
  File "./bessctl", line 137, in run_cli
    cli.loop()
  File "./bessctl", line 112, in loop
    super(BESSCLI, self).loop()
  File "/home/bess/bess/bessctl/cli.py", line 492, in loop
    self.process_one_line()
  File "/home/bess/bess/bessctl/cli.py", line 403, in process_one_line
    self.call_func(func, args)
  File "./bessctl", line 64, in call_func
    super(BESSCLI, self).call_func(func, args)
  File "/home/bess/bess/bessctl/cli.py", line 378, in call_func
    func(*args)
  File "/home/bess/bess/bessctl/commands.py", line 743, in run_conf
    _run_file(cli, conf_file, env_map)
  File "/home/bess/bess/bessctl/commands.py", line 735, in _run_file
    _do_run_file(cli, conf_file)
  File "/home/bess/bess/bessctl/commands.py", line 684, in _do_run_file
    code = compile(xformed, conf_file, 'exec')
  File "/home/bess/bess/bessctl/conf/my/bug.bess", line 1
    Source() -> Bypass() -> Sink()
              ^
SyntaxError: invalid syntax
```

**Comment by justinemarie** [comment]
This bug has caused me infinite grief as well, Felician! I'll talk with @sangjinhan a bit about it. Thanks for adding an issue.

**Comment by nemethf** [comment]
I'm not really confident about it, but the following patch might solve the general issue.  At least, it's a step in the right direction.  I hope.
```
diff --git a/bessctl/sugar.py b/bessctl/sugar.py
index bca88ee..1c34fe0 100644
--- a/bessctl/sugar.py
+++ b/bessctl/sugar.py
@@ -176,7 +176,9 @@ def replace_rarrows(s):
             last_token = token
 
     except (tokenize.TokenError, IndentationError):
-        return s    # source code with syntax errors. give up.
+        # Source code has syntax errors, but arrows has been set
+        # correctly up until now.
+        pass
 
     segments = []
     curr_seg = []
```
After this change, we should probably prepare to catch an exception in the other generate_tokens() call as well.

**Comment by sangjinhan** [comment]
Fixed in #490 (7d38119). Thank you!
